From 39daf472f954d8a0568912064adc9494d29ca2ed Mon Sep 17 00:00:00 2001
From: guanzi008 <20619190+guanzi008@users.noreply.github.com>
Date: Tue, 28 Jul 2026 15:55:00 +0800
Subject: [PATCH] AOSCOS: build with C++20

Apache Arrow 23 exposes std::span in its public Parquet headers. Scilab's
spreadsheet module includes those headers, so C++17 is no longer sufficient.
---
 configure.ac | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 7ae57b2..6828cbb 100644
--- a/configure.ac
+++ b/configure.ac
@@ -809,10 +809,10 @@ fi
 
 AM_CONDITIONAL(IS_64_BITS_CPU, test $IS_64_BITS_CPU = true)
 
-# check if the compiler supports c++17
+# check if the compiler supports c++20
 saved_cppflags="$CXXFLAGS"
 CXXFLAGS=""
-AX_CXX_COMPILE_STDCXX(17, noext, mandatory)
+AX_CXX_COMPILE_STDCXX(20, noext, mandatory)
 STDCXX_CXXFLAGS="$CXXFLAGS"
 CXXFLAGS="$saved_cppflags"
 
-- 
2.55.0

